internal/common/filespec.go

```diff
@@ -208,18 +213,61 @@ func (this *FileSpec) GetPath() string {
 
 // Copy the target of this FileSpec in the specified location.
 // If the FileSpec has a dest, then that will be the name of the resultant dirent within destDir.
-// If the filespec is a path, then copy a dirent.
+// If the filespec is a path, then copy all matching dirents.
 // If the filespec is a git repo, then ensure it is cloned/updated.
 // Empty and Nil FileSpecs are no-ops.
 // |onlyContents| applies to paths that are dirs and insists that only the contents of dir
 // and not the base dir itself is copied.
+// |baseDir| provides the relative base.
 func (this *FileSpec) CopyTarget(baseDir string, destDir string, onlyContents bool) error {
 	switch this.Type {
 	case FILESPEC_TYPE_EMPTY, FILESPEC_TYPE_NIL:
 		// no-op.
 		return nil
 	case FILESPEC_TYPE_PATH:
-		return this.copyPath(baseDir, destDir, onlyContents)
+		if !filepath.IsAbs(this.Path) && (baseDir != "") {
+			this.Path = filepath.Join(baseDir, this.Path)
+		}
+
+		destPath := ""
+		if onlyContents {
+			if this.Dest == "" {
+				destPath = destDir
+			} else {
+				destPath = filepath.Join(destDir, this.Dest)
+			}
+
+		} else {
+			filename := this.Dest
+			if filename == "" {
+				filename = filepath.Base(this.Path)
+			}
+
+			destPath = filepath.Join(destDir, filename)
+		}
+
+		paths, err := this.matchTargets()
+		if err != nil {
+			return fmt.Errorf("Failed to match target(s) for path '%s': '%w'.", this.Path, err)
+		}
+
+		// Create a new directory if multiple targets are matched.
+		if !util.PathExists(destPath) && len(paths) > 1 {
+			err := util.MkDir(destPath)
+			if err != nil {
+				return fmt.Errorf("Failed to make a directory: '%v'.", err)
+			}
+		}
+
+		// Loop over each matched path and copy it to the destination.
+		for _, path := range paths {
+			err := copyPath(path, destPath, onlyContents)
+			if err != nil {
+				return fmt.Errorf("Failed to copy target '%s': '%w'.", path, err)
+			}
+		}
+
+		return nil
```

Comment:
Move to a function.
I especially don't like putting a lot of code in a conditional (doubly so for a swtich/case (which just seems like a looser if).

You can also consider merging with matchTargets(), since that got smaller.